RF docs

AndrewWilliams3142 opened the pull request:
Added an example notebook for the CRM emulation with Random Forest. Also added a `utils_rf.py` which contains some functions I use for plots and downloading data from Zenodo.

Let me know if you want any other changes :)

duncanwp commented on examples/utils_rf.py:
```diff
@@ -0,0 +1,131 @@
+    ax.set_ylabel('Prediction')
+    
+    
+def LeaveOneOut(Xdata, Ydata, rndseed=0, **rf_kwargs):
```
This is really cool! I think it could be really useful for the other models too though, would you mind making it take the model as a parameter to make it more general? You could also consider using the existing scikit-learn [LeaveOneOut](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.LeaveOneOut.html) interface :-)

AndrewWilliams3142 replied:
I did try the LeaveOneOut interface from scikit but had a bit of trouble using it for more complex array types... I've made this function more general though, it now takes a `model` argument and can also do a baseline linear regression - because why not? haha

duncanwp commented on examples/utils_rf.py:
```diff
@@ -0,0 +1,131 @@
+
+    return output
+
+def get_crm_data(cache_path='.', preprocess=True):
```
I think this is probably the only function that belongs in /examples, could you merge this one with examples/utils.py and the rest with GCEm/utils.py

AndrewWilliams3142 replied:
Done!

duncanwp commented:
I really like the notebook, but on reflection it's probably overkill just to demonstrate the RFModel. Would just having the example #2 and the plot in cell 23 simplify things?

AndrewWilliams3142 commented:
> I really like the notebook, but on reflection it's probably overkill just to demonstrate the RFModel. Would just having the example #2 and the plot in cell 23 simplify things?

Could do! It depends what you want really. I'll do these changes and then we can see what it looks like 👍

AndrewWilliams3142 commented:
@duncanwp this is done! 

- [x] Added new, less overkill notebook
- [x] merged utils
- [x] added a `model` argument to the `LeaveOneOut` function and moved it to `GCEm.utils`

duncanwp commented on GCEm/utils.py:
```diff
@@ -137,6 +165,72 @@ def ensemble_collocate(ensemble, observations, member_dimension='job'):
+                model_ = estimators[model](training_params=X_train, 
+                                           training_data=Y_train, 
+                                           random_state=rndseed, 
+                                           **rf_kwargs)
```
Could you change `rf_kwargs` to `model_kwargs` and then add random_state to it (if it's not already in there), that way you can lose this if/else block :-)

AndrewWilliams3142 replied:
Good idea!

AndrewWilliams3142 replied:
@duncanwp done!

duncanwp commented:
Looks awesome thanks for putting this together :-)
